### PR TITLE
Add test cases of loader.go

### PR DESCRIPTION
**Pull request by kt3k**
This PR adds the test cases for error situations of `awsiotprotocol/loader`. This should improve the coverage score.

### CODE REVIEW

File: awsiotprotocol/loader_test.go [hunk 1]
```diff
@@ -16,6 +16,8 @@ package awsiotprotocol
 
 import (
 	"testing"
+
+	"github.com/stretchr/testify/assert"
```
Comment (at-wat) [inline awsiotprotocol/loader_test.go]: Personally, I like Golang's testing strategy. https://golang.org/doc/faq#testing_framework

File: awsiotprotocol/loader_test.go [hunk 2]
```diff
@@ -33,3 +35,13 @@ func TestByURL(t *testing.T) {
 		}
 	}
 }
+
+func TestByURLFailure(t *testing.T) {
+	_, err := ByURL("@@://@@@/@@.@@@")
+	assert.NotNil(t, err, "Should return error when a malformed url is given")
+	assert.Contains(t, err.Error(), "parse @@://@@@/@@.@@@: ")
```
Comment (at-wat) [inline awsiotprotocol/loader_test.go]: `assert.NotNil` seems to continue running the remaining test even if the assertion fails.
If `ByURL` returns nil, this line causes a segmentation fault.
(I guess this is one of the typical cases what https://golang.org/doc/faq#testing_framework is intended.)

File: awsiotprotocol/loader_test.go (hunk 1 above)
Comment (kt3k) [reply awsiotprotocol/loader_test.go]: OK. I'll update the test cases.

File: awsiotprotocol/loader_test.go [hunk 3]
```diff
@@ -33,3 +34,22 @@ func TestByURL(t *testing.T) {
...
+	}{
+		{input: "@@://@@@/@@.@@@", expectedErrorMessage: "parse @@://@@@/@@.@@@: "},
+		{input: "https://non-supported.protocol.com", expectedErrorMessage: "Protocol \"https\" is not supported"},
+	}
+	for _, v := range testcase {
+		_, err := ByURL(v.input)
+		if err == nil {
+			t.Errorf("awsiotprotocol.ByURL should fail with invalid input.\ninput: %#v", v.input)
+		}
+		if !strings.Contains(err.Error(), v.expectedErrorMessage) {
```
Comment (at-wat) [inline awsiotprotocol/loader_test.go]:
```suggestion
		else if !strings.Contains(err.Error(), v.expectedErrorMessage) {
```
`err.Error()` causes segmentation fault if `err == nil`

Comment (kt3k) [reply awsiotprotocol/loader_test.go]: Ah, now I understand your previous comment and what go faq says. These cases should never crash during the test run (even when test fails). Thank you for the suggestion!

**Pull request merged**